### PR TITLE
Add SQL source/sink

### PR DESCRIPTION
**Pull request by willcrichton**

### CODE REVIEW

File: scripts/travis-build.sh [hunk 1]
```diff
@@ -28,10 +28,10 @@ build_docker() {
         docker build -t $DOCKER_REPO:$1-local . \
                --build-arg gpu=OFF --build-arg tag=cpu --build-arg deps_opt='' \
                -f docker/Dockerfile.scanner
-        # travis_wait allows tests to run for N minutes with no output
-        # https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
+
+        # We run the tests as non-root user because Postgres test uses initdb which requires not being root
         docker run $DOCKER_REPO:$1-local /bin/bash \
-               -c "cd /opt/scanner/build && CTEST_OUTPUT_ON_FAILURE=1 make test"
+               -c "adduser --disabled-password --gecos \"\" user && su -c \"cd /opt/scanner/build && CTEST_OUTPUT_ON_FAILURE=1 make test\" user"
```
Comment (apoms) [inline scripts/travis-build.sh]: Oh god.

File: stdlib/misc/sql_source.cpp [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+          sql_config.hostaddr(), sql_config.port(), sql_config.dbname(),
+          sql_config.user(), sql_config.password())});
+    } catch (pqxx::pqxx_exception& e) {
+      LOG(FATAL) << e.base().what();
+    }
+  }
+
+  i64 total_elements() override {
+    try {
+    pqxx::work txn{*conn_};
```
Comment (apoms) [inline stdlib/misc/sql_source.cpp]: Indentation?

File: stdlib/misc/sql_source.cpp [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+      pqxx::oid type = field.type();
+      std::string& typname = pq_types_[type];
+      bool assigned = false;
+
+      #define ASSIGN_IF(NAME, TYPE) \
+        if (typname == NAME) { jrow[name] = field.as<TYPE>(); assigned = true; }
+
+      ASSIGN_IF("int4", i32);
+      ASSIGN_IF("float8", f32);
```
Comment (apoms) [inline stdlib/misc/sql_source.cpp]: Does this only support ints and floats? What about strings?

File: tests/py_test.py [hunk 4]
```diff
@@ -1278,3 +1279,77 @@ def test_job_timeout(timeout_db):
...
+def test_sql(sql_db):
+    (db, sql_config, cur) = sql_db
+
+    sql_query = db.protobufs.SQLQuery(
+        fields='test.id as id, test.num as num',
+        table='test',
+        id='test.id',
+        group='test.id')
+
+    row = db.sources.SQL(config=sql_config, query=sql_query)
```
Comment (apoms) [inline tests/py_test.py]: Could we document somewhere what the output format of `sources.SQL` is and the input format of `sink.SQL` is? As a comment is fine, but you could also put it here http://scanner.run/api/stdlib.html#built-in-sources

File: stdlib/misc/sql_source.cpp (hunk 3 above)
Comment (willcrichton) [reply stdlib/misc/sql_source.cpp]: Added support for more data types

File: scripts/travis-build.sh (hunk 1 above)
Comment (willcrichton) [reply scripts/travis-build.sh]: 😢

**Comment by apoms** [comment]
Looks good to me!

**Pull request merged**